### ISSUE TITLE
Error during install

### ISSUE THREAD

**Issue by michaelwittmann** [dlr-eoc/pgh3#2]
Running make in pgh3 leads to following error:

```gcc -I/usr/local/include/ -L/usr/local/lib/ -DEXTVERSION='"0.2.0"' -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/hierarchy.o src/hierarchy.c
gcc -I/usr/local/include/ -L/usr/local/lib/ -DEXTVERSION='"0.2.0"' -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/neighbor.o src/neighbor.c
src/neighbor.c: In function ‘h3_kring’:
src/neighbor.c:70:9: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
         for (int i = 0; i < max_num_neighbors; i++) {
         ^
src/neighbor.c:70:9: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code
<builtin>: recipe for target 'src/neighbor.o' failed
make: *** [src/neighbor.o] Error 1
```

Any ideas how to fix it?

**Comment by nmandery** [comment]
Hi Michael,

are you using a gcc version < 5.1.0? Looking at [this stackoverflow post](https://stackoverflow.com/questions/14737104/what-is-the-default-c-mode-for-the-current-gcc-especially-on-ubuntu#14737642) it seems the versions before 5.1.0 used the C90 standard which did not allow the dclaration in the for loop I used. Or are you using a completely different compiler?

To resolve this issue you may add `-std=c11` to the pgh3 Makefile to the `override CFLAGS` line - hopefuly your compiler supports C11, otherwise try `-std=c99`.

I will also add it to the Makefile in the future.

**Comment by michaelwittmann** [comment]
I was using gcc 4.9.2.
Overriding CFLAGS in Makefile fixed it!
Thanks for the fast reply!

**Comment by michaelwittmann** [comment]
Hi Nico,
trying to install version 0.2.1 produced a new error on my machine:
```
XYZ:~/pgh3$ make
gcc -I/usr/local/include/ -L/usr/local/lib/ -DEXTVERSION='"0.2.1"' -std=c11 -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/util.o src/util.c
src/util.c: In function ‘__h3_polyfill_palloc0’:
src/util.c:126:45: error: too many arguments to function ‘GetConfigOptionByName’
         const char * max_polyfill_mem_str = GetConfigOptionByName(PGH3_POLYFILL_MEM_SETTING_NAME, NULL, true);
                                             ^
In file included from src/util.c:23:0:
/usr/include/postgresql/9.5/server/utils/guc.h:369:14: note: declared here
 extern char *GetConfigOptionByName(const char *name, const char **varname);
              ^
src/util.c:131:72: error: ‘GUC_UNIT_MB’ undeclared (first use in this function)
             if (!parse_int(max_polyfill_mem_str, &max_polyfill_mem_mb, GUC_UNIT_MB, NULL)) {
                                                                        ^
src/util.c:131:72: note: each undeclared identifier is reported only once for each function it appears in
<builtin>: recipe for target 'src/util.o' failed
make: *** [src/util.o] Error 1
```

**Comment by nmandery** [comment]
Hi Michael,

seems the function signature of `GetConfigOptionByName` changed with PostgreSQL 9.6: https://github.com/postgres/postgres/commit/10fb48d66de76e7dc1e36ef18af502ed9600352f

Also `GUC_UNIT_MB` was only added in PostgreSQL 10: https://github.com/postgres/postgres/commit/9a3215026bd6955e88bd8c20542cfe6acffdb1c8

I use mostly Postgres > 10 for my development and deployments, so with the recent changes I thing this extension just supports this version. I need to add this to the documentation, and will check sometime in the next few days if I can resolve these two issues. At least for the `GetConfigOptionByName` issue it will not be easy to find a workaround for.

I think I need to add a CI environment to test this extension against multiple postgresql versions :)

**Comment by michaelwittmann** [comment]
Thanks Nico,

We are fine right now with version 0.2.0. 
I think it is worth considering the support of versions <10, as still many old versions are running out there. Postgres still lists 9.5 as an offficial supported version.

I will keep following the updates. Thanks a lot for your immediate response!

**Comment by nmandery** [comment]
Hi Michael,

of course you are right. I also prefer it to work on the 9.x versions, although I tend to just remove the memory configuration feature for these versions. I will create a new release soon.

Thank you for your inputs.

**Comment by nmandery** [comment]
Although it is not particularly elegant, this issue should be resolved with commit 2fad7c3
